File: Code/new_fitervari_backend/src/main/java/at/htl/control/TrainerRepository.java
package at.htl.control;

import at.htl.model.Trainer;
import io.quarkus.hibernate.orm.panache.PanacheRepository;

import javax.enterprise.context.ApplicationScoped;

@ApplicationScoped
public class TrainerRepository implements PanacheRepository<Trainer> {

    public Trainer save(Trainer entity) {
        return getEntityManager().merge(entity);
    }
}

File: Code/fitervari_backend/src/main/java/at/htl_leonding/repository/TrainerRepository.kt
package at.htl_leonding.repository

import at.htl_leonding.model.Trainer
import io.quarkus.hibernate.orm.panache.PanacheRepository
import javax.enterprise.context.ApplicationScoped
import javax.inject.Inject
import javax.persistence.EntityManager

@ApplicationScoped
class TrainerRepository : PanacheRepository<Trainer> {
    @Inject
    lateinit var em: EntityManager

    fun save(trainer: Trainer): Trainer {
        return em.merge(trainer)
    }
}